<a href="https://colab.research.google.com/github/s-hub404/DS-ML/blob/main/Problem_solving/Advance_distributions.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [3]:
import numpy as np
import pandas as pd
import scipy.stats as sc

Problem:

0.02% of credit card holders of a company report theft of their credit cards each month. The company has a total 15,000 credit cards.

a. What is the probability that no one reports theft of credit cards during the next month?

b. Determine the expected number of reported stolen credit cards in a month.

In [11]:
# This problem relates to Poisson's dist. as we want No. of events during a period.

mu = 0.0002*15000
# constant rate of occurnace

# for no reports of theft , k = 0
k = 0
prob_1 = sc.poisson.cdf(k, mu)
print('Probability a: ',prob_1)

ans_b = 0.0002*15000
print('Answer part b: ', ans_b)

Probability a:  0.04978706836786395
Answer part b:  3.0


Theoritical explanation:

Population proportion, p = 0.0002
Sample size, n = 15000

Let “x” represents the number of cards stolen then the mean number of cards stolen per 15000 cards: λ=np=3

a. The probability that no one reports the loss or theft of his or her credit cards: P(x=0) = (e−3)*(30)/0! ≅ 0.0498

b. The expected number of reported lost or stolen credit cards, μ=λ=3.

Problem:

During a pandemic, a pharmaceutical company was able to successfully engineer a drug to be able to fight the virus. But, the probability of manufacturing the drug successfully was 40% on each independent production process. Based on this data, what would be the probability of successfully manufacturing the drug in less than 7 production processes?

In [16]:
# This is also a case of Geometric distributions.
# The geometric distribution describes the probability of experiencing 
# a certain amount of failures before experiencing the first success in a series of Bernoulli trials.

p = 0.4

# Prob. of successfully manufacture a drug in less than 7 trials.
# i.e we have 6 failures before a success.
k = 6
prob = sc.geom.cdf(k, p)
prob

0.953344

Theoretical explanation:

Given, probability of success = 40%
p = 0.4
q = 1-p=0.6
It’s asked, what is the probability of succesfully manufacturing in less than 7 trials
So, P(X < 7)
We can understand from the problem statement that it’s a geometric distribution.
Now,
P(X < 7) = P(X = 1) + P(X = 2) + ….. P(X = 6)
= p * q(1-1) + p * q(2-1) + …… p * q(6-1)

